packages/mobile/src/screens/chat-screen/ChatUserListScreen.tsx

```diff
@@ -0,0 +1,274 @@
+    },
+    debounceMs,
+    [query, setHasQuery, dispatch]
+  )
+
+  const handleChange = useCallback(
+    (text: string) => {
+      setQuery(text)
+    },
+    [setQuery]
+  )
+
+  const handleLoadMore = useCallback(() => {
+    if (status === Status.LOADING || defaultUserList.loading) {
+      return
+    }
+    if (hasQuery) {
+      dispatch(searchUsers({ query }))
+    } else {
+      defaultUserList.loadMore()
+    }
+  }, [hasQuery, query, status, defaultUserList, dispatch])
+
+  const handlePress = (item) =>
+    navigation.navigate('Chat', { chatId: item.chat_id })
+
+  const renderItem = ({ item: user, index }: { item: User; index: number }) => {
+    if (!user) {
+      return <LoadingSpinner />
+    }
+
+    return (
+      <TouchableHighlight onPress={() => handlePress(user)}>
+        <View style={styles.userContainer}>
+          <ProfilePicture profile={user} style={styles.profilePicture} />
+          <View style={styles.userNameContainer}>
+            <UserBadges user={user} nameStyle={styles.userName} />
+            <Text style={styles.handle}>@{user.handle}</Text>
+            <View style={styles.followContainer}>
+              <View style={styles.followersContainer}>
+                <IconUser
+                  fill={palette.neutralLight4}
+                  height={styles.iconUser.height}
+                  width={styles.iconUser.width}
+                />
+                <Text style={styles.followersCount}>{user.follower_count}</Text>
+                <Text style={styles.followers}>{messages.followers}</Text>
+              </View>
+              {user.does_follow_current_user ? (
+                <Text style={styles.followsYouTag}>{messages.followsYou}</Text>
+              ) : null}
+            </View>
+          </View>
+        </View>
+      </TouchableHighlight>
+    )
+  }
+
+  return (
+    <Screen
+      url='/chat'
+      title={messages.title}
+      icon={IconCompose}
+      topbarRight={null}
+    >
+      <ScreenContent>
+        <View style={styles.rootContainer}>
+          <View style={styles.searchContainer}>
+            <TextInput
+              placeholder={messages.search}
+              Icon={() => (
+                <IconSearch
+                  fill={palette.neutralLight4}
+                  width={styles.searchIcon.width}
+                  height={styles.searchIcon.height}
+                />
+              )}
+              styles={{
+                root: styles.searchInputContainer,
+                input: styles.searchInputText
+              }}
+              onChangeText={handleChange}
+              value={query}
+            />
+          </View>
+
+          {users.length > 0 ? (
+            <FlatList
+              onEndReached={handleLoadMore}
+              data={users}
+              renderItem={renderItem}
+              keyExtractor={(user) => user.user_id}
+            />
+          ) : (
+            <LoadingSpinner style={styles.loadingSpinner} />
```

Comment:
I think I have this on web too, but this technically doesn't allow for an empty state. Something we'll probably want to ask about

Reply:
Yep good call.